backend/fms_core/services/experiment_run_info.py

```diff
@@ -0,0 +1,353 @@
+                capture_details = _get_capture_details(derived_sample)
+                row.capture_kit = capture_details['capture_kit']
+                row.capture_baits = capture_details['capture_baits']
+            elif library.library_selection.name == "ChIP-Seq":
+                row.chip_seq_mark = library.library_selection.target
+
+    return row
+
+def _convert_flowcell_coordinate_to_lane_number(coord: str, container: Container) -> int:
+    '''Converts a freezeman coordinate (eg. A01) to an integer flowcell lane number.'''
+
+    container_spec = CONTAINER_KIND_SPECS[container.kind]
+    if container_spec is None:
+        raise Exception(f'Cannot convert coord {coord} to lane number. No ContainerSpec found for container kind "{container.kind}".')
+
+    lane = convert_alpha_digit_coord_to_ordinal(coord, container_spec.coordinate_spec)
+    return lane
+
+def _find_library_prep(library: Library) -> Union[ProcessMeasurement, None]:
+    ''' 
+    Given a library, find the Library Preparation process that was used to create the
+    library, if any, to extract the library preparation properties (eg. Library Kit).
+
+    Note that not all libraries will have a Library Preparation measurement,
+    since they can be imported directly into freezeman without passing the
+    Library Preparation step.
+    '''
+    
+    library_prep: Union[ProcessMeasurement, None] = None
+    try:
+        library_to_find : Library = library
+
+        if library.library_selection is not None:
+            # If a library was captured then that generated a new library instance.
+            # We have to find the previous instance, on which the Library Prep was run.
+            try:
+                capture_protocol = Protocol.objects.get(name="Library Capture")
+                capture_lineage: SampleLineage = SampleLineage.objects.get(
+                    process_measurement__process__protocol_id=capture_protocol.pk,
+                    child__derived_samples__library__id=library.pk
+                )
+                #library_to_find = Library.objects.get(derived_sample__biosample_id=derived_sample.biosample.id)
+                # Now find the derived sample in the parent that contains the same biosample id as our library.
+                library_to_find = Library.objects.get(derived_sample__samples=capture_lineage.parent, derived_sample__biosample=library.derived_sample.biosample)
+                
+            except:
+                pass
+
+        # Find the library preparation process measurement
+        protocol: Protocol = Protocol.objects.get(name="Library Preparation")
+        lineage: SampleLineage = SampleLineage.objects.get(
+            process_measurement__process__protocol_id=protocol.pk,
+            child__derived_samples__library__id=library_to_find.pk)
+        library_prep = lineage.process_measurement
+    except SampleLineage.DoesNotExist:
+        pass
+    except SampleLineage.MultipleObjectsReturned:
+        pass
```

Comment:
please don't pass on exception handling ... even if it is a bit of a performance hit test with .exist() first then if it does do the get(). This is one more hit on the DB but it is much less risky. i also doubt it is a normal occurrence that we get a MultipleObjectReturned... we would not get this to pass by silently (although i think it would crash anyway but then why try to catch it). Given the peculiar get() in django it could make sense for performance to use the except pass, but not much for the multipleobjectreturned.

Reply:
Yeah, I wasn't sure what to do with MultipleObjectsReturned - I'll raise an exception instead.
I need to let DoesNotExist pass, though, because not all libraries have a library prep step. For one of my experiments I submitted libraries in a pool, so they did not go through Prep or QC, and thus there is no Library Preparation process for them (and thus, no Library Kit value for the run info). I'll refactor the function to make this clearer.